Incorporate Windows-based CI (AppVeyor et al)

Previous attempt https://github.com/servo/servo/pull/9406

Hrm, it looks like heapsize is still failing with:
```
note: C:\Users\appveyor\AppData\Local\Temp\1\rustc.YtB1bF5ggNNd\libheapsize-da165bc9088f0d78.rlib(heapsize-da165bc9088f0d78.0.o): In function `heapsize::heap_size_of':
C:/projects/servo/.cargo\registry\src\github.com-88ac128001ac3a9a\heapsize-0.2.5\src/lib.rs:41: undefined reference to `je_malloc_usable_size'
collect2.exe: error: ld returned 1 exit status
 
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
```

cc @Jayflux

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,33 @@
+version: 1.0.{build}
+
+branches:
+  only:
+    - master
+
+platform:
+  - x64
+
+cache:
+  - .servo
+  - .cargo
+
+install:
+  - set MSYSTEM=MSYS
+  - set MSYS="winsymlinks=lnk"
+  - PATH C:\msys64\mingw64\bin;C:\msys64\usr\bin\;%PATH%
+  - bash -lc "echo $MSYSTEM; pacman --needed --noconfirm -Sy pacman-mirrors"
+  - bash -lc "pacman --noconfirm -Sy"
+  - bash -lc "pacman -Sy --needed --noconfirm git mingw-w64-x86_64-toolchain mingw-w64-x86_64-freetype mingw-w64-x86_64-icu mingw-w64-x86_64-nspr mingw-w64-x86_64-ca-certificates mingw-w64-x86_64-expat mingw-w64-x86_64-cmake tar diffutils patch patchutils make python2-setuptools"
+  - bash -lc "easy_install-2.7 pip virtualenv"
+  - bash -lc "mv /mingw64/bin/python2.exe /mingw64/bin/python2-mingw64.exe"
+  - bash -lc "mv /mingw64/bin/python2.7.exe /mingw64/bin/python2.7-mingw64.exe"
+
+build_script:
+  - cmd: >-
+      set MSYSTEM=MINGW64
+
+      PATH C:\msys64\mingw64\bin;C:\msys64\usr\bin\;%PATH%
+
+      bash -lc "cd $APPVEYOR_BUILD_FOLDER; ./mach build -d -v"
+
+test: off

--- a/components/profile/lib.rs
+++ b/components/profile/lib.rs
@@ -8,6 +8,7 @@
 #![feature(plugin)]
 #![plugin(plugins)]
 
+#[cfg(not(target_os = "windows"))]
 extern crate alloc_jemalloc;
 extern crate hbs_pow;
 extern crate ipc_channel;

--- a/components/profile/mem.rs
+++ b/components/profile/mem.rs
@@ -463,6 +463,7 @@ mod system_reporter {
                    newp: *mut c_void, newlen: size_t) -> c_int;
     }
 
+    #[cfg(not(target_os = "windows"))]
     fn jemalloc_stat(value_name: &str) -> Option<usize> {
         // Before we request the measurement of interest, we first send an "epoch"
         // request. Without that jemalloc gives cached statistics(!) which can be
@@ -496,6 +497,11 @@ mod system_reporter {
         }
 
         Some(value as usize)
+    }
+
+    #[cfg(target_os = "windows")]
+    fn jemalloc_stat(value_name: &str) -> Option<usize> {
+        None
     }
 
     // Like std::macros::try!, but for Option<>.

